python/paddle/geometric/sampling/neighbors.py

```diff
@@ -170,3 +170,151 @@ def sample_neighbors(
     if return_eids:
         return out_neighbors, out_count, out_eids
     return out_neighbors, out_count
+
+
+def weighted_sample_neighbors(
+    row,
+    colptr,
+    edge_weight,
+    input_nodes,
+    sample_size=-1,
+    eids=None,
+    return_eids=False,
+    name=None,
+):
+    """
+    Graph Weighted Sample Neighbors API.
+
+    This API is mainly used in Graph Learning domain, and the main purpose is to
+    provide high performance of graph weighted-sampling method. For example, we get the
+    CSC(Compressed Sparse Column) format of the input graph edges as `row` and
+    `colptr`, so as to convert graph data into a suitable format for sampling, and the
+    input `edge_weight` should also match the CSC format. Besides, `input_nodes` means
+    the nodes we need to sample neighbors, and `sample_sizes` means the number of neighbors
+    and number of layers we want to sample. This API will finally return the weighted sampled
+    neighbors, and the probability of being selected as a neighbor is related to its weight,
+    with higher weight and higher probability.
```

Comment:
Randomly sample neighbor nodes based on edge weights.

rewriting of this text by ChatGPT:

This API is primarily used in the domain of Graph Learning, with the main purpose of providing high-performance graph weighted-sampling methods. For example, it takes the input graph edges in Compressed Sparse Column (CSC) format as `row` and `colptr`, which allows for converting the graph data into a suitable format for sampling. The `edge_weight` input should also match the CSC format. Additionally, the `input_nodes` parameter specifies the nodes from which we need to sample neighbors, while `sample_sizes` determines the number of neighbors and layers to sample. Finally, this API returns the weighted sampled neighbors, where the probability of being selected as a neighbor is influenced by its weight, with higher weight resulting in higher probability.